tests/lib/prepare.sh

```diff
@@ -518,3 +529,266 @@ prepare_all_snap() {
 
     disable_kernel_rate_limiting
 }
+
+setup_reflash_magic_core18() {
+        # install the stuff we need
+        distro_install_package kpartx busybox-static
+        distro_install_local_package "$GOHOME"/snapd_*.deb
+        distro_clean_package_cache
+
+        snap install "--${CORE_CHANNEL}" core18
+        snap install "--${CORE_CHANNEL}" snapd
+        # install ubuntu-image
+        snap install --classic --edge ubuntu-image
+
+        # needs to be under /home because ubuntu-device-flash
+        # uses snap-confine and that will hide parts of the hostfs
+        IMAGE_HOME=/home/image
+        mkdir -p "$IMAGE_HOME"
+
+        # modify the snapd snap so that it has our snapd
+        UNPACKD="/tmp/snapd-snap"
+        unsquashfs -d "$UNPACKD" /var/lib/snapd/snaps/snapd_*.snap
+        dpkg-deb -x "$SPREAD_PATH"/../snapd_*.deb "$UNPACKD"
+        snap pack "$UNPACKD" "$IMAGE_HOME"
+
+        # FIXME: fetch directly once its in the assertion service
```

Comment:
What prevents this currently?

Reply:
We need to clarify how to specify the kernel for core18, see https://forum.snapcraft.io/t/how-to-specific-the-kernel-snap-on-core18/5947